core/src/main/java/org/apache/calcite/sql/fun/SqlInternalOperators.java

```diff
@@ -224,6 +224,16 @@ private SqlInternalOperators() {
   public static final SqlAggFunction LITERAL_AGG =
       SqlLiteralAggFunction.INSTANCE;
 
+  public static final SqlOperator CAST_NOT_NULL =
```

Comment:
I think you should document in the JavaDoc how this operator differs from a typical CAST, copying some of the discussion above.